starter-code/src/components/ContactsList.js

```diff
@@ -0,0 +1,75 @@
+import React, { Component } from "react";
+import Contact from "./Contact";
+
+class ContactsList extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      contacts: props.contacts.splice(0, 5),
+      sort: "none"
+    };
+  }
+
+  deleteContactHandler = contactIndex => {
+    const contactsCopy = this.state.contacts;
+    contactsCopy.splice(contactIndex, 1);
+    this.setState({
+      contacts: contactsCopy
+    });
+  };
+
+  addContact = () => {
+    let rand = Math.floor(Math.random() * this.props.contacts.length);
+    let contact = this.props.contacts[rand];
+    const contactsNew = this.state.contacts;
+    contactsNew.unshift(contact);
+    this.setState({ contacts: contactsNew });
+  };
+
+  sortByName = () => {
+    let sortedByName = this.state.contacts.sort((a, b) => {
```

Comment:
.sort() mutates the array in place. Protect your array against mutations ;)

Reply:
👍